gammapy/utils/parallel.py

```diff
@@ -0,0 +1,89 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Multiprocessing and multithreading setup"""
+import logging
+from gammapy.utils.pbar import progress_bar
+
+log = logging.getLogger(__name__)
+
+
+MULTIPROCESSING_BACKEND = "multiprocessing"
+N_PROCESSES = 1
+N_THREADS = 1
+
+
+def get_multiprocessing(backend=None):
+    """import multiprocessing module for a given backend"""
+
+    if backend is None:
+        backend = MULTIPROCESSING_BACKEND
+    if backend == "multiprocessing":
+        import multiprocessing
+
+        return multiprocessing
+    elif backend == "ray":
+        import ray.util.multiprocessing as multiprocessing
+
+        return multiprocessing
+    else:
+        raise ValueError("Invalid multiprocessing backend")
+
+
+def run_multiprocessing(
+    func,
+    inputs,
+    backend=None,
+    pool_kwargs=None,
+    method="starmap",
+    method_kwargs=None,
+    task_name="",
+):
+    """Run function in a loop or in parralel"""
+
+    if method not in ["starmap", "apply_async"]:
```

Comment:
Maybe add a test for the incorrect method.

Reply:
done